src/reynir_correct/pattern.py

```diff
@@ -470,6 +849,171 @@ def create_patterns(cls) -> None:
                     None,
                 )
             )
+            # Catch ""
+            p.append(
+                (
+                    "vitni",  # Trigger lemma for this pattern
+                    "VP > { VP > 'verða' NP-PRD > { 'vitni' PP > { P > { 'af' } } } }",
+                    cls.wrong_preposition_vitni_af,
+                    None,
+                )
+            )
+            # Catch "Hún gerði grín af því."
+            p.append(
+                (
+                    "grín",  # Trigger lemma for this pattern
+                    "VP > { VP > [ .* 'gera' .* "
+                    'NP-OBJ > "grín" ] ( PP | ADVP ) > ( { P > { "af" } } | { "af" } ) } ',
+                    cls.wrong_preposition_grin_af,
+                    None,
+                )
+            )
+            # Catch "Þetta er mesta vitleysa sem ég hef gert grín af."
+            p.append(
+                (
+                    "grín",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > { VP > { 'gera' } NP-OBJ > { 'grín' } } } ADVP > { 'af' } }",
+                    cls.wrong_preposition_grin_af,
+                    None,
+                )
+            )
+            # Catch "...og gerir grín af sjálfum sér."
+            p.append(
+                (
+                    "grín",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > 'gera' NP-OBJ } PP > { 'af' } }",
+                    cls.wrong_preposition_grin_af,
+                    None,
+                )
+            )
+
+            p.append(
+                (
+                    "grín",  # Trigger lemma for this pattern
+                    "VP > { PP > { NP > { 'grín' } } PP > { 'af' } }",
+                    cls.wrong_preposition_grin_af,
+                    None,
+                )
+            )
+            # Catch "Hann leiðir líkur af því."
+            p.append(
+                (
+                    "leiða",  # Trigger lemma for this pattern
+                    "VP > { VP > { VP > { 'leiða' } NP-OBJ > { ('líkur' | 'rök') } } PP > { P > { 'af' } } }",
+                    cls.wrong_preposition_leida_af,
+                    None,
+                )
+            )
+            # Catch "Hann leiðir ekki líkur af því."
+            p.append(
+                (
+                    "líkur",  # Trigger lemma for this pattern
```

Comment:
With this trigger lemma, this rule never catches 'leiða rök'

Reply:
Ah of course. I fixed this and made 'leiða' the trigger lemma for all three search patterns.